src/Engines/thinEngine.ts

```diff
@@ -4700,6 +4700,83 @@ export class ThinEngine {
         return internalFormat;
     }
 
+    /**

```

Comment:
Could this be moved from thinEngine to prevent growing it?

Reply:
I first put it in Misc/TextureTools.ts, but it means a dependency on this file from Extensions/engine.readTexture.ts: is that not a problem?

Reply:
so why not placing it in Extensions/engine.readTexture.ts ?

Reply:
but texture tools would be ok

Reply:
Moved to TextureTools.

Reply:
Ok, it does not work to put it in `TextureTools`, so I have moved the code to `engine.readTexture`.